src/pages/HomePage/HomePage.tsx

```diff
@@ -1,5 +1,66 @@
-import React from 'react';
+import React, { useState } from 'react';
+import SearchBar from '../../components/SearchBar/SearchBar';
+import { HomepageContext } from '../../context/GlobalContext';
 
 export default function HomePage() {
-  return <div className="page">[Home page]</div>;
+  const [searchTerm, setSearchTerm] = useState('');
+  const [recipesList, setRecipesList] = useState([]);
+  const [searchError, setSearchError] = useState('');
+
+  const limit = 1;
+  const baseUrl = process.env.REACT_APP_API_BASE_RECIPES_URL;
+  const apiKey = process.env.REACT_APP_API_KEY;
+  const searchUrl = `${baseUrl}/search?apiKey=${apiKey}&number=${limit}&query=${searchTerm}`;
+  const randomUrl = `${baseUrl}/random?apiKey=${apiKey}&number=${limit}`;
+
+  const getSearchData = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    try {
+      const response = await fetch(searchUrl);
+      const jsonData = await response.json();
+      setRecipesList(jsonData.results);
+      setSearchTerm('');
+    } catch (error) {
+      setSearchError(error);
+    }
+  };
+
+  const getRandomData = async () => {
+    try {
+      const response = await fetch(randomUrl);
+      const jsonData = await response.json();
+      setRecipesList(jsonData.recipes);
+    } catch (error) {
+      setSearchError(error);
+    }
+  };
+
+  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    setSearchTerm(e.target.value);
+  }
+
+  return (
+    <HomepageContext.Provider
```

Comment:
I Would personally add context into App.tsx or even in Index.tsx to wrap all application. Perhaps we'll use just one context so we can add also login details. But we can also use separate context for it, but it's maybe not necessary.